File: backend/routers/note.py
from fastapi import APIRouter, HTTPException

from models.session_models import (
    NoteGenerateRequest,
    NoteGenerateResponse,
    NoteApproveRequest,
    NoteApproveResponse,
    NoteShareRequest,
    NoteShareResponse,
    PublicNoteResponse,
    SectionSummaryRequest,
)
from services.note_service import generate_note_for_session, approve_note_for_session, share_note_for_session, generate_section_summary
from services.session_service import get_session

router = APIRouter(prefix="/api/note", tags=["note"])

@router.post("/generate", response_model=NoteGenerateResponse)
def generate_note(request: NoteGenerateRequest):
    return generate_note_for_session(request)

@router.post("/approve", response_model=NoteApproveResponse)
def approve_note(request: NoteApproveRequest):
    return approve_note_for_session(request)

@router.post("/share", response_model=NoteShareResponse)
def share_note(request: NoteShareRequest):
    return share_note_for_session(request)

@router.post("/section-summary", response_model=NoteGenerateResponse)
def generate_section_summary_endpoint(request: SectionSummaryRequest):
    return generate_section_summary(request)


@router.get("/public/{session_id}", response_model=PublicNoteResponse)
def get_public_note(session_id: str):
    session = get_session(session_id)
    if session is None:
        raise HTTPException(status_code=404, detail='세션을 찾을 수 없습니다.')
    if not session.public_note:
        raise HTTPException(status_code=404, detail='공개 노트가 없습니다. 먼저 공유를 생성하세요.')
    return PublicNoteResponse(public_note=session.public_note)

File: frontend/src/api/note.ts
const baseUrl = import.meta.env.VITE_API_BASE_URL || `${window.location.protocol}//${window.location.hostname}:8000`

export interface NoteGenerateRequest {
  session_id: string
}

export interface NoteGenerateResponse {
  note: string
}

export interface NoteApproveRequest {
  session_id: string
  note: string
}

export interface NoteApproveResponse {
  approved: boolean
}

export interface NoteShareRequest {
  session_id: string
}

export interface NoteShareResponse {
  public_note: string
}

export interface PublicNoteResponse {
  public_note: string
}

export async function generateNote(request: NoteGenerateRequest): Promise<NoteGenerateResponse> {
  const response = await fetch(`${baseUrl}/api/note/generate`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(request),
  })

  if (!response.ok) {
    throw new Error('노트 생성에 실패했습니다.')
  }

  return response.json()
}

export async function approveNote(request: NoteApproveRequest): Promise<NoteApproveResponse> {
  const response = await fetch(`${baseUrl}/api/note/approve`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(request),
  })

  if (!response.ok) {
    throw new Error('노트 승인에 실패했습니다.')
  }

  return response.json()
}

export async function shareNote(request: NoteShareRequest): Promise<NoteShareResponse> {
  const response = await fetch(`${baseUrl}/api/note/share`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(request),
  })

  if (!response.ok) {
    throw new Error('공유 노트 생성에 실패했습니다.')
  }

  return response.json()
}

export async function fetchPublicNote(session_id: string): Promise<PublicNoteResponse> {
  const response = await fetch(`${baseUrl}/api/note/public/${session_id}`)
  if (!response.ok) {
    throw new Error('공개 노트를 불러오지 못했습니다.')
  }

  return response.json()
}
